### PR TITLE
✨ Use .NET OpenAPI and Scalar

### PR DESCRIPTION
**Pull request by MattParkerDev**
﻿> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

#383 

> 2. What was changed?

Removed nswag, added scalar
Cleaned up unnecessary http launch profiles

> 3. Did you do pair or mob programming?

no

image
Figure: Scalar UI

### CODE REVIEW

**Review by danielmackay — COMMENTED** [review]
ADR Missing

**Review by danielmackay — COMMENTED** [review]
C# should be the default Rest client example

**Comment by MattParkerDev** [comment]
> C# should be the default Rest client example

It is?

File: src/WebApi/DependencyInjection.cs [hunk 1]
```diff
@@ -13,9 +13,9 @@ public static void AddWebApi(this IServiceCollection services, IConfiguration co
     {
         services.AddHttpContextAccessor();
         services.AddScoped<ICurrentUserService, CurrentUserService>();
-
-        services.AddOpenApiDocument(configure => configure.Title = "CleanArchitecture API");
+        
         services.AddEndpointsApiExplorer();
```
Comment (danielmackay) [inline src/WebApi/DependencyInjection.cs:17]: Can you check if we definitely need the call to `AddEndpointsApiExplorer()` ?

File: src/WebApi/Program.cs [hunk 2]
```diff
@@ -25,16 +26,13 @@
     app.UseHsts();
 }
 
+app.MapOpenApi();
```
Comment (danielmackay) [inline src/WebApi/Program.cs:29]: `Program.cs` is pretty clean at the moment.  Do you think we should wrap the Open API stuff up into an extension method?

**Comment by danielmackay** [comment]
> > C# should be the default Rest client example
> 
> It is?

You're right.  Ignore that.

File: src/WebApi/DependencyInjection.cs (hunk 1 above)
Comment (MattParkerDev) [reply src/WebApi/DependencyInjection.cs:17]: .AddOpenApi() adds it internally, so not necessary anymore. Will remove.

image

Comment (MattParkerDev) [reply src/WebApi/DependencyInjection.cs]: Done.

File: src/WebApi/Program.cs (hunk 2 above)
Comment (MattParkerDev) [reply src/WebApi/Program.cs:29]: 🤷‍♂️ 1 line vs 2 currently. Happy to if you wish. I don't think it is necessary.

Comment (danielmackay) [reply src/WebApi/Program.cs:29]: OK.  Let's leave for now

**Review by danielmackay — APPROVED** [review]
LGTM

**Pull request merged**